clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```
@@ -1085,18 +1062,18 @@ private void ensureExplicitAcknowledgement() {
      */
     private static AcknowledgementMode initializeAcknowledgementMode(ConsumerConfig config, Logger log) {
         if (config == null) {
-            return AcknowledgementMode.UNKNOWN;
+            return AcknowledgementMode.IMPLICIT;
         }
-        String acknowledgementModeStr = config.getString(ConsumerConfig.INTERNAL_SHARE_ACKNOWLEDGEMENT_MODE_CONFIG);
-        if ((acknowledgementModeStr == null) || acknowledgementModeStr.isEmpty()) {
-            return AcknowledgementMode.UNKNOWN;
-        } else if (acknowledgementModeStr.equalsIgnoreCase("implicit")) {
+        String acknowledgementModeStr = config.getString(ConsumerConfig.SHARE_ACKNOWLEDGEMENT_MODE_CONFIG);
+        if ((acknowledgementModeStr == null) ||
+            acknowledgementModeStr.isEmpty() ||
+            acknowledgementModeStr.equalsIgnoreCase("implicit")) {
             return AcknowledgementMode.IMPLICIT;
         } else if (acknowledgementModeStr.equalsIgnoreCase("explicit")) {
             return AcknowledgementMode.EXPLICIT;
         }
-        log.warn("Invalid value for config {}: \"{}\"", ConsumerConfig.INTERNAL_SHARE_ACKNOWLEDGEMENT_MODE_CONFIG, acknowledgementModeStr);
-        return AcknowledgementMode.UNKNOWN;
+        log.warn("Invalid value for config, {}: \"{}\", setting mode to default value : IMPLICIT", ConsumerConfig.SHARE_ACKNOWLEDGEMENT_MODE_CONFIG, acknowledgementModeStr);
+        return AcknowledgementMode.IMPLICIT;
```

For the case when the config had an invalid value, the code previously logged a `WARN` log and returned `UNKNOWN`, now that we do not have `UNKNOWN` mode anymore, I can think of 2 options here:
1. To throw an `IllegalArgumentException`, as the user should only enter "implicit" or "explicit"
2. Log a warning and just assume the default value of `IMPLICIT`.
I have kept the 2nd option now, but I do feel in case of invalid config entry, we need to throw an exception so that the user will correct the entry. For empty/null values we can still assume `IMPLICIT`, but maybe we can throw exception here?